### PR TITLE
Add missing Request properties and WebSocket support 

### PR DESCRIPTION
**Pull request by mikemaccana**
- [Y] Use a meaningful title for the pull request. Include the name of the package modified.

- [Y] Test the change in your own code. (Compile and run.)
- [N] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [Y] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [Y] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [Y] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [Y] Provide a URL to documentation or source code which provides context for the suggested changes: https://v6.arc.codes/primitives/http#req

### CODE REVIEW

File: types/architect__functions/ws.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        connectionId: string;
+    };
+    isBase64Encoded: boolean;
+    body: string;
+}
+
+// WebSockets still respond with HTTP
+// (usually just a 200 acknowledging the socket is connected)
+export interface WebSocketHandler {
+    (req: WebsocketEvent): Promise<HttpResponse | undefined>;
```
Comment (activescott) [inline types/architect__functions/ws.d.ts:16]: Is it a full `HttpResponse` here? I actually don't know, but if it is a subset of `HttpResponse` properties I'd suggest a more specific type.

File: types/architect__functions/ws.d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+export interface ArcWs {
```
Comment (activescott) [inline types/architect__functions/ws.d.ts:19]: I suggest adding some "tests" for this in [/types/architect__functions/test](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/architect__functions/test). Note the "tests" don't need to exercise any behavior, just use the types. So they're usually pretty trivial. [This section in readme](https://github.com/DefinitelyTyped/DefinitelyTyped/#my-package-teststs) explains a bit more.

**Review by activescott — CHANGES_REQUESTED** [review]
Thanks for the add @mikemaccana !! I left a couple comments. I don't know about the types on the ws so that was really a question. i do think we should have tests for these to make sure we keep them table as others make prs in the future. Let me know if I can help in any way!

Also looks like some checks failed with some lint issues. Those will need fixed before the bot will merge it :)

File: types/architect__functions/ws.d.ts (hunk 2 above)
Comment (mikemaccana) [reply types/architect__functions/ws.d.ts:19]: Ah that makes sense - I did look at the tests but couldn't see anything actually being tested. I'll finish this off soon.

**Comment by mikemaccana** [comment]
Guys I started a new role with a YC startup two months ago and haven't had much time for Arc work since, sorry about this. If someone picks adding the extra properties (according to the poster, adding a unit test that doesn't test anything and simply uses them is considered sufficient) we could land this.

**Pull request closed without merging**